content/doc/book/pipeline/reference.adoc

```diff
@@ -0,0 +1,492 @@
+---
+layout: section
+---
+:notitle:
+:description:
+:author:
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+:toc:
+:imagesdir: /doc/book/resources
+:hide-uri-scheme:
+
+= Pipeline DSL reference
```

Comment:
I think this document should be called "Declarative Reference" very explicitly. Calling it "Pipeline DSL Reference" means, IMO, that it should contain a *lot* more (covering Scripted and Declarative Pipeline) than I think is prudent to add to the Handbook,


That said, I agree that this should exist